iaso/diffing/differ.py

```diff
@@ -40,6 +48,8 @@ def load_pyramid(self, version, validation_status=None, top_org_unit=None, org_u
             queryset = queryset.hierarchy(parent)
         if org_unit_types:
             queryset = queryset.filter(org_unit_type__in=org_unit_types)
+        if org_unit_group:
+            queryset = queryset.filter(groups=org_unit_group).distinct()  # distinct because groups is m2m
```

Comment:
```suggestion
            queryset = queryset.filter(groups=org_unit_group).distinct("id")  # distinct because groups is m2m
```

I'd be specific in the distinct clause to avoid comparing every fields.